src/main/java/com/synacy/gradprogram/spock/exercise/RefundService.java

```diff
@@ -4,14 +4,36 @@
 
 public class RefundService {
 
-  public BigDecimal calculateRefund() {
-    // TODO: Implement me. Full refund if cancel reason is due to damaged item.
-    //  Also full refund if the order was cancelled within 3 days of order date, else refund half of the total cost.
-    return null;
+  private final RefundRepository refundRepository;
+
+  public RefundService(RefundRepository refundRepository) {
+    this.refundRepository = refundRepository;
+  }
+
+  public BigDecimal calculateRefund(CancelOrderRequest request, Order order) {
+    long refundDateLimit = order.getDateOrdered().getTime() + (86400000 * 3);
+    BigDecimal refundAmount;
+
+    if (request.getReason() == CancelReason.DAMAGED || request.getDateCancelled().getTime() < refundDateLimit) {
+      refundAmount = BigDecimal.valueOf(order.getTotalCost());
+    } else {
+      refundAmount = BigDecimal.valueOf(order.getTotalCost()/2);
+    }
+
+    createAndSaveRefundRequest(order, refundAmount);
```

Comment:
This is not part of the scope/responsibility of the method and in the instructions.